ert/data/record/_transformation.py

```diff
@@ -227,3 +229,69 @@ async def _save_record_to_file(record: Record, location: Path, mime: str) -> Non
     else:
         async with aiofiles.open(str(location), mode="wb") as fb:
             await fb.write(record.data)  # type: ignore
+
+
+class RecordTreeTransformation(RecordTransformation):
+    """
+    Write all leaf records of a NumericalRecordTree to individual files.
+    """
+
+    def __init__(self, sub_path: Optional[str] = None):
+        if sub_path is not None:
+            raise NotImplementedError("Extracting sub-trees not implemented")
+
+    async def transform_input(
+        self,
+        record: Record,
+        mime: str,
+        runpath: Path,
+        location: Path,
+    ) -> None:
+        if not isinstance(record, NumericalRecordTree):
+            raise TypeError("Only NumericalRecordTrees can be transformed.")
+        for key, leaf_record in record._flat_record_dict.items():
```

Comment:
This uses a private method. We could consider making it public, but I'm not so sure we should expose this as part of the interface for the record :thinking:  Maybe its ok for the RecordTreeTransformation to roll its own flatten, functionality?

Reply:
There's a public property called `flat_record_dict`, which could be used instead ...

Reply:
Oh yes :)